InputState input_state
---
ActivationState activation_state